analytics.js

```diff
@@ -533,9 +533,16 @@ class Analytics {
     if (typeof from === "object") (options = from), (from = null);
 
     const rudderElement = new RudderElementBuilder().setType("alias").build();
+    const updatedFrom =
+      typeof from === "string" || typeof from === "undefined"
+        ? from
+        : JSON.stringify(from);
     rudderElement.message.previousId =
-      from || (this.userId ? this.userId : this.getAnonymousId());
-    rudderElement.message.userId = to;
+      updatedFrom || (this.userId ? this.userId : this.getAnonymousId());
+    rudderElement.message.userId =
+      typeof to === "string" || typeof to === "undefined"
+        ? to
+        : JSON.stringify(to);
```

Comment:
There is a good amount of redundant logic in this PR. Let's refactor.

Reply:
Addressed